src/System.Runtime.Numerics/tests/ComplexTests.cs

```diff
@@ -354,12 +361,10 @@ public static IEnumerable<object[]> ASin_Advanced_TestData()
             yield return new object[] { 0, 1234000000, 0, 21.62667394298955 }; // Imaginary part is positive
 
             // Boundary values
-            yield return new object[] { double.MaxValue, 0, double.NaN, double.NaN };
-            yield return new object[] { double.MinValue, 0, double.NaN, double.NaN };
-            yield return new object[] { 0, double.MaxValue, double.NaN, double.NaN };
-            yield return new object[] { 0, double.MinValue, double.NaN, double.NaN };
-            yield return new object[] { double.MaxValue, double.MaxValue, double.NaN, double.NaN };
-            yield return new object[] { double.MinValue, double.MinValue, double.NaN, double.NaN };
+            // We used to test that MinValue and MaxValue produced NaNs, but that's an implementation artifact and
+            // mathematically incorrect. For example, asin(Double.MaxValue) ~ 1.5708 \pm 710.47 i. In fact,
+            // for all arguments in the repesentable range, asin and acos lie in the representable range.
+            // We should fix the implementation so it returns the right values.
```

Comment:
Removing comments. I can't add the correct assertions for Asin yet because the Asin method hasn't been fixed. (It still computes z*z explicitly, and so passes infinities into Sqrt.) I do intend to fix Asin and Acos too, but I'm trying to contain the scope of this PR for now.

These comments touch on the thorniest issue with this PR: it changes the behavior of Sqrt for infinities and NaNs. For example, previously Sqrt mapped (Infinity,0)->(Infinity,NaN), with this change it maps (Infinity,0)->(Infinity,0). Now the new behavior is actually more sensible, and there were no tests asserting that it should be any particular way, but it was the way it was, and if anyone relied on that (God help him!), this would break him.

I think it would be helpful to have a discussion about exactly what we behavior we expect for infinities and NaNs for the complex functions, and to codify the results as test assertions. But this is quite a complicated issue: there are lots of combinations, separate real and complex parts, and performance trade-offs if algorithms must do checks to enforce them. Really doing this carefully would vastly expand the scope of this check-in.

What I am proposing is: until proper infinity/NaN behavior is codified, accept PRs that change it, as long as they don't make it manifestly less sensible, under the theory that no one should be relying on random, unprincipled infinity/NaN behavior anyway.

@hughbe: I now know how to skip a test for netfx, but I don't know how to add one for netfx only.